pycytominer/cyto_utils/sqlite/convert.py

```diff
@@ -0,0 +1,456 @@
+    sql_engine: Engine:
+        SQLite database engine
+    column_data: List[dict]:
+        Column metadata from database
+    prepend_tablename_to_cols: bool:
+        Determines whether we prepend table name
+        to column name.
+    avoid_prepend_for: list:
+        List of column names to avoid tablename prepend
+    chunk_list_dicts: list:
+        List of dictionaries for chunked querying
+    filename: str:
+        Filename to be used as prefix parquet export
+
+    Returns
+    -------
+    str
+        Filename of parquet file created.
+    """
+
+    # gather a set of the table names
+    # note: coldata contains repeated tablenames and we gather unique
+    # tablenames only via set from the list.
+    table_list = set([coldata["table_name"] for coldata in column_data])
+
+    # build empty dataframe which will become the basis of concat operations
+    concatted = pd.DataFrame()
+
+    for table in table_list:
+        # query data from table by join key chunks, ensuring unique column names where necessary
+        to_concat = sql_table_to_pd_dataframe.run(
+            sql_engine=sql_engine,
+            table_name=table,
+            prepend_tablename_to_cols=prepend_tablename_to_cols,
+            avoid_prepend_for=avoid_prepend_for,
+            chunk_list_dicts=chunk_list_dicts,
+            column_data=column_data,
+        )
+
+        # if concatted is empty, we set it to the first dataframe
+        if len(concatted) == 0:
+            concatted = to_concat
+
+        # else we concat the existing dataframe with the new one
+        else:
+            # both the already concatted and target are prepared with matching columns nan's
+            # for data they do not contain.
+            concatted = nan_data_fill.run(fill_into=concatted, fill_from=to_concat)
+            to_concat = nan_data_fill.run(fill_into=to_concat, fill_from=concatted)
+
+            # bring the prepared dataframes together as the new concatted
+            concatted = pd.concat([concatted, to_concat])
+
+    # export concatted result from all tables to a uniquely name parquet file
+    filename_uuid = to_unique_parquet.run(df=concatted, filename=filename)
+
+    # return the filename generated from to_unique_parquet
+    return filename_uuid
+
+
+@task
+def to_unique_parquet(df: pd.DataFrame, filename: str) -> str:
+    """
+    Write a uniquely named parquet from provided dataframe.
+
+    Parameters
+    ----------
+    df: pd.DataFrame:
+        dataframe to use for parquet write
+    filename: str:
+        filename to use as a prefix along with uuid
+
+    Returns
+    -------
+    str
+        Unique filename for parquet file created.
+    """
+
+    # gather unique id for filename
+    file_uuid = str(uuid.uuid4().hex)
+
+    # build a unique filename string
+    unique_filename = f"{filename}-{file_uuid}.parquet"
+
+    # export the dataframe based on the unique filename
+    df.to_parquet(unique_filename, compression=None)
```

Comment:
Great comment, thanks for bringing this up. It might be worth another code comment to help illustrate the "why", either way. I used `compression=None` here as files generated by this function are temporary serialized chunks which are later brought together as one file (outside of this function). My intent was to conserve time/performance with faster writes for these chunks. As a result, we could run into read performance drawbacks. Performance is likely to hinge on the chunk size, dataset differences, and the machine running the code.

Based on this, do you feel we should add a parameter for compression options? We could also lean on the Pandas default.

Reply:
After further testing on this front, I found there is little difference between using `compression=None` or not. As a result, I've committed a change to remove this parameter for the parquet chunks to avoid specializing where we seemingly don't benefit from it.